### DISCUSSION TITLE
Scheduled Jobs

### DISCUSSION THREAD

**Discussion by skuflow** [oqtane/oqtane.framework#3059] [Q&A]
how do you add a scheduled job in oqtane 4?  There is no add button in the UI like there is in the old tutorial video on youtube.

**Comment by leigh-pointer** [comment]
@skuflow look inside the oqtane.server project there is a folder called jobs the notifications job is there which you can use as a guide.

**Reply by skuflow** [reply]
let me clarify my question - how do you deploy it?  for example, you can upload themes and modules.

**Reply by leigh-pointer** [reply]
@skuflow there is no separate deployment system for server components! Currently you would need to create a module and deploy that way.

**Reply by sbwalker** [reply]
@skuflow take a look at the DNF.Projects repo - it contains a module which contains a scheduled job. Scheduled jobs are automatically discovered during startup - they do not need to be registered manually (which is why there is no "add job" option in the UI)
